src/tqec/computation/correlation.py

```diff
@@ -169,3 +170,31 @@ def external_stabilizer(self, io_ports: list[int]) -> str:
             for port in io_ports
         ]
         return "".join(paulis)
+
+    def external_stabilizer_on_graph(self, graph: BlockGraph) -> str:
+        """Get the Pauli operator supported on the input/output ports of the graph.
+        The ports are ordered according to their labels.
+
+        Args:
+            g: The block graph to consider.
+
+        Returns:
+            The Pauli operator supported on the input/output ports of the graph.
+        """
+        port_labels = graph.ordered_ports
+        port_positions = [graph.ports[p] for p in port_labels]
+        zx = graph.to_zx_graph()
+        zx_ports = [zx.p2v[p] for p in port_positions]
+        return self.external_stabilizer(zx_ports)
+
+    def area(self) -> int:
+        """Return the area of the correlation surface.
+
+        The area of the correlation surface is the number of nodes it spans.
+        An X node and a Z node with the same id are counted as two nodes.
+        """
+        span_nodes = {node for edge in self.span for node in edge}
+        return len(span_nodes)
+
+    def __add__(self, other: CorrelationSurface) -> CorrelationSurface:
+        return CorrelationSurface(self.span.symmetric_difference(other.span))
```

Comment:
For me, it would make more sense to have the `xor` (`^`) operator doing that:
```suggestion
    def __xor__(self, other: CorrelationSurface) -> CorrelationSurface:
        return CorrelationSurface(self.span.symmetric_difference(other.span))
```